### DISCUSSION TITLE
Testing OpenVINO Music Separation with Audacity 3.4.2

### DISCUSSION THREAD

**Discussion by petersampsonaudacity** [intel/openvino-plugins-ai-audacity#74] [Show and tell]
I have just tested this on a couple of my favourite test tracks:
a) track 1 of The Beatles Sgt Pepper 
b) Frank Sinatra's Come fly with me.  

With both of them the vocal separation is extremely impressive, very impressive indeed.  I just used 2-Stem on Sgt Pepper - but on Sinatra I tried 4-stem - here I expected to see more in the drums stem (but I need to listen more carefully before passing judgement).

I note that it takes a lot of processing time the 2 min Sgt Pepper too 2m30secs and the 3 min Sinatra took 4 mins - and it certainly makes by processor work hard (I used the default CPU setting) - not many things get the fan running on my HP Envy but this tool certainly does 😉

![image](https://github.com/intel/openvino-plugins-ai-audacity/assets/50205984/ead0045d-bf73-4757-82e6-5bc96261073e)

**Comment by petersampsonaudacity** [comment]
So I did some more drum testing with two instrumental drum heavy tracks:
c) Sussex By The Sea by The Band of H.M. Royal Marines
d) Sandy Nelson's Let There be Drums.

The drum separation in Sussex by the Sea was stunningly excellent and the vocal track totally silent.

Sandy Nelson was 99% good at the drums but still impressive - and had a small section of high pitched squeal in the vocal track - not vocals as this is an instrumental, but that is easily mixed back into the Other Instruments track.
![image](https://github.com/intel/openvino-plugins-ai-audacity/assets/50205984/8c96d6d6-e35d-41fb-b8c6-fe1082185526)

**Comment by RyanMetcalfeInt8** [comment]
@petersampsonaudacity -- Thank you for sharing!

> I note that it takes a lot of processing time the 2 min Sgt Pepper too 2m30secs and the 3 min Sinatra took 4 mins - and it certainly makes by processor work hard (I used the default CPU setting) - not many things get the fan running on my HP Envy but this tool certainly does 😉

Yes, this feature uses Demucs v4 AI model (developed by Meta), and is quite heavy. Depending on your system, and if 'GPU' shows up in the drop-down list of devices, GPU may produce results faster than CPU.

**Comment by petersampsonaudacity** [comment]
TL:DR folk who have yearned for good vocal separation are likely to love this. 😎

**Comment by Tserklaes** [comment]
Hello
The installation was straightforward.
But separation doesn't work.

Audacity error logs:
15:29:04: Operation 'OpenVINO Music Separation' took 0,025000 seconds. Poll was called 2 times and took 0,000001 seconds. Yield was called 0 times and took 0,000000 seconds.

**Comment by RyanMetcalfeInt8** [comment]
> Hello The installation was straightforward. But separation doesn't work.
> 
> Audacity error logs: 15:29:04: Operation 'OpenVINO Music Separation' took 0,025000 seconds. Poll was called 2 times and took 0,000001 seconds. Yield was called 0 times and took 0,000000 seconds.

Hi @Tserklaes, could you open a new issue with some more details that include more details of your system setup (CPU version, Windows version, etc.). As well as some more details about what you mean by "doesn't work" (screenshots help alot here).

Thanks,
Ryan

**Comment by Meowdoleon** [comment]
> @petersampsonaudacity -- Thank you for sharing!
> 
> > I note that it takes a lot of processing time the 2 min Sgt Pepper too 2m30secs and the 3 min Sinatra took 4 mins - and it certainly makes by processor work hard (I used the default CPU setting) - not many things get the fan running on my HP Envy but this tool certainly does 😉
> 
> Yes, this feature uses Demucs v4 AI model (developed by Meta), and is quite heavy. Depending on your system, and if 'GPU' shows up in the drop-down list of devices, GPU may produce results faster than CPU.

Yup, can confirm that a three minutes 45 seconds song took me about three minutes to separate using an RX 580 GPU, and it did wonder. You can faintly hear the vocal in the instrumental and vice versa, but I think that is expected, especially from a song using synthesized voice as vocals.

**Comment by techtalktoll** [comment]
Tested separation using 4 on a variety of pieces - all with vocals with total success. Why would I do it on   backing tracks when I am trying to make backing tracks??? Added my vocals and they were separated successfully. I was using a USB audio box 48k and 24 bit with YouTube if this makes a difference.
